java/sage/PseudoMenu.java

```diff
@@ -121,6 +117,9 @@ public void activate(boolean redo)
     }
 
     multipleTextInputs = comp.getNumTextInputs(0) > 1;
+
+    // notify the UI Renderer as to which Menu we are activating, so that it can pass it to the client as a HINT
+    uiMgr.getRootPanel().getRenderEngine().setMenuHint(widg.getName(), null, comp.getNumTextInputs(0)>0);
```

Comment:
This can be better optimized so that getNumTextInputs is only called once and the value is stored and then reused to set multipleTextInputs and also the hint properly.